### PR TITLE
feat(96): init and create command

### PR DESCRIPTION
**Pull request by OrlovAlexei**
Resolves: https://github.com/statoscope/statoscope/issues/96

- feat(cli): added create and init rule
- fix(webpack-model): added initial field in type Chunk
- feat(cli): added init command
- feat(cli): added create command # Please, use semantic commit message

### CODE REVIEW

File: packages/cli/src/commands/templates/reporter.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { FileExt } from '.';
+
+export function reporterTemplate(fileExt: FileExt): string {
+  return fileExt === FileExt.ts ? reporterTs : reporterJs;
+}
+
+const reporterJs = `class NewReporter {
+  constructor(){}
+
+  async run(result){
```
Comment (smelukov) [inline packages/cli/src/commands/templates/reporter.ts:10]:
```suggestion
  async run(result) {
```

File: packages/cli/src/commands/templates/reporter.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { FileExt } from '.';
+
+export function reporterTemplate(fileExt: FileExt): string {
+  return fileExt === FileExt.ts ? reporterTs : reporterJs;
+}
+
+const reporterJs = `class NewReporter {
+  constructor(){}
```
Comment (smelukov) [inline packages/cli/src/commands/templates/reporter.ts:8]:
```suggestion
  constructor() {}
```

File: packages/cli/src/commands/templates/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { ruleTemplate } from './rule';
+import { FileExt } from '.';
+
+export function pluginTemplate(fileExt: FileExt): string {
+  return fileExt === FileExt.ts ? asyncChunkRulePluginTs : asyncChunkRulePluginJs;
+}
+
+export const asyncChunkRulePluginTs = `import type{ PluginFn } from '@statoscope/stats-validator/dist/plugin';
```
Comment (smelukov) [inline packages/cli/src/commands/templates/plugin.ts:8]:
```suggestion
export const asyncChunkRulePluginTs = `import type { PluginFn } from '@statoscope/stats-validator/dist/plugin';
```

File: packages/cli/src/commands/templates/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import type{ Rule } from '@statoscope/stats-validator/dist/rule';
```
Comment (smelukov) [inline packages/cli/src/commands/templates/plugin.ts:9]:
```suggestion
import type { Rule } from '@statoscope/stats-validator/dist/rule';
```

File: packages/cli/src/commands/templates/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import type{ Prepared } from '@statoscope/webpack-model';
```
Comment (smelukov) [inline packages/cli/src/commands/templates/plugin.ts:10]:
```suggestion
import type { Prepared } from '@statoscope/webpack-model';
```

File: packages/cli/src/commands/templates/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+${ruleTemplate(FileExt.ts, {
+  export: false,
+  import: false,
+})}
+const newStatoscopePlugin: PluginFn<Prepared> = () => {
+  return {
+    rules: {
+      'async-chunk': asyncChunkRule,
```
Comment (smelukov) [inline packages/cli/src/commands/templates/plugin.ts:19]: let's add some dummy-import for this

File: packages/cli/src/commands/templates/index.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  reporter: reporterTemplate,
+  plugin: pluginTemplate,
+} as const;
+
+export type TemplateName = keyof typeof mapper;
+
+export function getTemplate(templateName: TemplateName, fileExt: FileExt): string {
+  if (templateName === 'rule') {
+    return mapper.rule(fileExt, {
+      export: true,
```
Comment (smelukov) [inline packages/cli/src/commands/templates/index.ts:23]: seems like we can replace `import/export` with `module: 'commonjs'/'esm'`

File: packages/cli/src/commands/templates/index.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  rule: ruleTemplate,
+  reporter: reporterTemplate,
+  plugin: pluginTemplate,
+} as const;
+
+export type TemplateName = keyof typeof mapper;
+
+export function getTemplate(templateName: TemplateName, fileExt: FileExt): string {
+  if (templateName === 'rule') {
+    return mapper.rule(fileExt, {
```
Comment (smelukov) [inline packages/cli/src/commands/templates/index.ts:22]: I think that template function might be described as:
```ts
type TemplateOptions = {
  output: {
    type: 'js' | 'ts',
    module: 'commonjs' | 'esm',
  }
};
type Template<TAdditionalOptions> = (TemplateOptions, TAdditionalOptions) => string;
// TAdditionalOptions - is template-specific options, might be passed from cli args
```

Example:
```ts
const asyncChunkRuleTs: Template<unknown> = (options): string => options.output.type === 'ts' ? `...` : `...`
```

File: packages/cli/src/commands/create.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+const SUPPORTED_ENTITIES = ['plugin', 'rule', 'reporter'];
+const SUPPORTED_EXT = [FileExt.js, FileExt.ts];
+
+export default function (yargs: Argv): Argv {
+  return yargs.command(
+    'create [entity] [output]',
+    `Generate plugin, rule, reporter file
+Examples:
+default usage: create rule
```
Comment (smelukov) [inline packages/cli/src/commands/create.ts:14]:
```suggestion
create rule # create example-rule from the template
```

File: packages/cli/src/commands/create.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+Custom path: create plugin --output path/to/folder
```
Comment (smelukov) [inline packages/cli/src/commands/create.ts:15]:
```suggestion
create plugin --output path/to/folder # create example-plugin from the template and save it into a specific directory
```

File: packages/cli/src/commands/create.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+File ext: create rule --ext ts`,
```
Comment (smelukov) [inline packages/cli/src/commands/create.ts:16]:
```suggestion
create rule --ext ts # create example-rule from the template and save it as typescript source`,
```

**Review by smelukov — CHANGES_REQUESTED** [review]
Really thanks for the PR!
Please look at some comments above ☺️

**Comment by OrlovAlexei** [comment]
@smelukov  Hello, thank you for the suggestion! I added support for ```esm``` and ```commonjs``` modules for JS files. I only added a comment about ``` jora-queries ``` because it only returns the ``` unknown ``` type. Maybe there is a task to fix this?

**Comment by smelukov** [comment]
@OrlovAlexei are you done here?

**Comment by OrlovAlexei** [comment]
@smelukov yes, I'm done

**Comment by smelukov** [comment]
Thank!
Sorry for so long waiting.
I've got some rest )

**Pull request merged**